utilities/post_migration.py

```
@@ -214,6 +215,84 @@ def _read_marker() -> str | None:
+        except (SSHException, AuthenticationException, NoValidConnectionsError, ChannelException) as e:
+            LOGGER.warning(f"SSH failed for VM {vm_name}: {type(e).__name__}: {e} - retrying...")
+            return None
+
+    luks_devices: list[dict[str, Any]] | None = None
+    try:
+        for sample in TimeoutSampler(wait_timeout=timeout, sleep=retry_delay, func=_check_luks):
+            if sample is not None:
+                luks_devices = sample
+                break
+    except TimeoutExpiredError as e:
+        raise TimeoutExpiredError(f"Could not verify LUKS encryption on VM {vm_name} after {timeout}s") from e
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)